docs/scos/dev/front-end-development/202212.0/oryx/reactivity/signals.md

```diff
@@ -0,0 +1,118 @@
+# Oryx Framework - Signals
+
+## Overview
+
+Signals are a crucial tool within the Oryx framework, offering a clean and efficient reactivity API for components. They allow for the declarative formulation of component logic while seamlessly integrating with observables from domain services.
+
+## Implementation of Signals in Oryx
+
+The Oryx implementation of signals has a core mechanism and a simplified API. The core is well-suited for advanced usage, while the simplified API is sufficiently robust for most components. This documentation will focus on the simplified API.
+
+### Creating Signals
+
+To create a signal, use the `signal()` function. This function can either take a raw value or accept an observable.
+
+Here's an example of creating a simple signal:
+
+```ts
+const counter = signal(1);
+```
+
+Changing the signal value:
+
+```ts
+counter.set(2);
+```
+
+Creating a signal from an observable:
+
+```ts
+const values = signal(observable$);
+```
+
+You can initialize signals with options to adjust their behavior:
+
+- `equal` function: This option allows for a custom equality function between two consecutive signal values. By default, strict comparison is used. Implementing your own function gives control over when a signal updates, avoiding unnecessary updates and performance costs when new and old values are practically identical.
+
+- `initialValue` option: This is used when creating a signal from an observable. It sets the first value of the signal, so you don't have to wait for the observable to give a value.
+
+Here's an example of using options:
+
+```ts
+const values = signal(observable$, { initialValue: 1, equal: (a, b) => a === b });
+```
+
+### Computed Signals
+
+A computed signal derives its value from other signals. It automatically reevaluates its value when any of the signals it depends on changes.
+
+Here's an example of a computed signal:
+
+```ts
+const counter = computed(() => 3 * counter(1));
+```
+
+Computed signals can also convert observables to signals transparently:
+
+```ts
+const counter = computed(() => productService.get({ sku: productSku() }));
+```
+
+In the example above, `productSku` is a signal, and `productService.get` returns an observable.
+
+Computed signals can use the same set of options as regular signals.
+
+### Effects
+
+Effects are functions that run whenever a signal's value changes.
+
+Here's an example:
+
+```ts
+const counter = effect(() => {
+    console.log('counter changed', counter());
+});
+```
+
+You can configure effects using options to modify their behavior:
+
+- `defer`: If this is set to true, your effect won't run until you explicitly call the `start()` method.
+- `async`: Set this to true if you want your effect to run asynchronously.
+
+Example:
+
+```ts
+const counter = effect(() => {
+    console.log('counter changed', counter());
+}, { defer: true, async: true });
+```
+
+## Using Signals in Components
+
+### @signalAware directive
```

Comment:
If you decide to remove the `## Using signals in components` section, then change header of this section from `###` to `##`
```suggestion
### @signalAware directive
```

Reply:
Do you mean also dropping 
```
## Implementation of Signals in Oryx
```
and raise all ### to ##?

Reply:
"## Implementation of Signals in Oryx" has a description so you can leave it

Reply:
Decided to add description also to the second ##, as removing it would break the whole structure (separate directives are not on the same level as Singals in Oryx)